Tournaments

javieroc opened the pull request.

javieroc committed:
Agregado tipos de torneos como relacion en torneos

javieroc committed:
Endpoints para obtener los tipos de torneos

mattgaviota commented on stapi/database/seeds/TournamentsTypesTableSeeder.php:
```diff
@@ -1,6 +1,7 @@
 
 use Illuminate\Database\Seeder;
+use Carbon\Carbon;
 
```
Si ya tenes el modelo es mejor para mi hacer otro tipo de seeder.
Algo como
``` php
<?php

use Illuminate\Database\Seeder;
use App\TournamentType;

class TournamentsTypesTableSeeder extends Seeder
{
    /**
     * Run the database seeds.
     *
     * @return void
     */
    public function run()
    {
        $data = [
            ['name' => 'league'],
            ['name' => 'cup']
        ];
        TournamentType::truncate();

        foreach ($data as $tournamentType) {
            TournamentType::create($tournamentType);
        }
    }
}
```
Entonces eso te permite agregar mas tipos facilmente, y te ahorras el carbon.

mattgaviota commented on stapi/routes/api.php:
```diff
@@ -25,5 +25,7 @@ function () {
         Route::resource('tournaments', 'TournamentController');
 
         Route::resource('teams', 'TeamController');
+
```
Acá deberías dejar más limpias las rutas usando https://laravel.com/docs/5.4/controllers#restful-partial-resource-routes

mattgaviota commented on stapi/app/Transformers/TournamentTransformer.php:
```diff
@@ -62,4 +62,18 @@ public function includeTeams(Tournament $tournament)
+    public function includeTournamentType(Tournament $tournament)
+    {
+        $tournamentType = $tournament->tournament_type;
+        //dd($tournamentType->id);
```
Estos comentarios hay que eliminarlos directamente.

mattgaviota reviewed (commented):
Ahí comenté un par de cosas para revisar

javieroc committed:
Mejoras en el codigo

 - Mejorar seeder para tipos de torneos
 - Remover linea comentada
 - Limpiar rutas, dejamos solo las necesarias
